libp2p/stream/bufferstream.nim

```diff
@@ -130,6 +147,12 @@ method readOnce*(s: BufferStream,
 
   s.activity = true
 
+  # We want to return 0 exactly once - after that, we'll start raising instead -
+  # this is a bit nuts in a mixed exception / return value world, but allows the
+  # consumer of the stream to rely on the 0-byte read as a "regular" EOF marker
+  # (instead of _sometimes_ getting an exception).
```

Comment:
Not sure this is necesary? Why not just raise?

Reply:
consistency? zero-return is customary, also because you if you have a fully regular communication, you should not be hitting _exceptions_ - it's not an exception that a channel is closed

Reply:
other streams _sometimes_ return 0 as well, this is mainly giving it a stable semantic that should be implemented across the board so that it's possible to reason about code flows somehow (and not having lots of different behaviours randomly)